src/live/emitters.js

```diff
@@ -0,0 +1,15 @@
+const { io } = require("../expressServer");
+
+function sendOrderCancelledEvent(ordersToUpdate) {
+  console.log("ordersToUpdate: ", ordersToUpdate);
+  
+  for (const userData of ordersToUpdate) {
+    const userId = userData._id.toString();
+    const numOrders = userData.orders.length;
+    const orderNoun = numOrders === 1 ? 'order' : 'orders';
+    
+    io.to(userId).emit('orders:cancelled', `${numOrders} of your ${orderNoun} have been cancelled.`);
+  }
```

Comment:
![](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/91bcdb3d-20b1-4fda-9f10-efb215d64522)

Here we get that circular dependency I warned of, when we start the app. So `io` will be undefined and this message won't get emmitted :( 

This is where we want to use the `io` variable that we stuck in the `global` object:

```suggestion
function sendOrderCancelledEvent(ordersToUpdate) {
  console.log("ordersToUpdate: ", ordersToUpdate);
  
  for (const userData of ordersToUpdate) {
    const userId = userData._id.toString();
    const numOrders = userData.orders.length;
    const orderNoun = numOrders === 1 ? 'order' : 'orders';
    
    global.io.to(userId).emit('orders:cancelled', `${numOrders} of your ${orderNoun} have been cancelled.`);
  }
```